bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/ui/wizard/CreateContractInputFromBusinessObjectWizardPage.java

```diff
@@ -244,23 +253,107 @@ private void createProcessDataMappingTreeViewer(final Composite composite, final
         inputTypeTreeViewerColumn.getColumn().setText(Messages.inputType);
         inputTypeTreeViewerColumn.getColumn().setWidth(150);
         inputTypeTreeViewerColumn.setLabelProvider(new InputTypeColumnLabelProvider());
-
         dbc.bindValue(ViewersObservables.observeInput(treeViewer),
                 selectedDataObservable,
                 null,
                 updateValueStrategy().withConverter(selectedDataToFieldMappings()).create());
-        dbc.addValidationStatusProvider(new MultiValidator() {
+        final IViewerObservableSet checkedElements = ViewersObservables.observeCheckedElements(treeViewer, FieldToContractInputMapping.class);
+        final WritableValue checkedObservableValue = new WritableValue();
+        checkedObservableValue.setValue(checkedElements);
+        final WritableValue mappingsObservableValue = new WritableValue();
+        mappingsObservableValue.setValue(fieldToContractInputMappingsObservable);
+        final MultiValidator multiValidator = createEmptySelectionMultivalidator(checkedElements);
+        dbc.addValidationStatusProvider(multiValidator);
+        dbc.bindValue(checkedObservableValue, mappingsObservableValue,
+                updateValueStrategy().withConverter(new Converter(IObservableSet.class, WritableList.class) {
```

Comment:
Extract Converter as a dedicated class, it will help to  write unit test on it and simplify code